edgelet/iotedge-proxy/src/settings.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::path::{Path, PathBuf};
+
+use config::{Config, ConfigError, File, FileFormat};
+use failure::Fail;
+use serde_derive::Deserialize;
+use url::Url;
+use url_serde;
+
+use crate::{Error, ErrorKind};
+
+pub const DEFAULTS: &str = include_str!("../config/default.yaml");
+
+pub const DEFAULT_SETTINGS_FILEPATH: &str = "/etc/iotedge-proxy/config.yaml";
+
+const TOKEN_FILEPATH: &str = "/var/run/secrets/kubernetes.io/serviceaccount/token";
```

Comment:
Can the user configure the location of the token in Kubernetes? If so, we should take that as config.

Reply:
Yes. User can override it from the config file. If it is empty there, k8s location will be used.